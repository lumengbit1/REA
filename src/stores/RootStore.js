import { configure } from 'mobx';

import FormStore from './FormStore/FormStore';

configure({ enforceActions: 'always' });

class RootStore {
    constructor() {
        this.formStore = new FormStore(this);
    }
}

export default RootStore;
