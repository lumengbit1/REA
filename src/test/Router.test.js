import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Root from '../router/Routers';
import RootStore from '../stores/RootStore';
import { Provider } from 'mobx-react';

const rootStore = new RootStore();
describe('Route Test', () => {
    it('Route Test', () => {
        const component = renderer
            .create(
                <Provider rootStore={rootStore}>
                    <MemoryRouter>
                        <Root />
                    </MemoryRouter>
                </Provider>
            )
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
