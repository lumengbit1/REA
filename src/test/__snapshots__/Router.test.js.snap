// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route Test Route Test 1`] = `
<div>
  <div
    className="App"
  >
    <div
      className="homePage"
    >
      <div
        className="resultsArea"
      >
        <div
          className="title"
        >
          Results
        </div>
        <div
          className="block"
        />
      </div>
      <div
        className="savedPropertiesArea"
      >
        <div
          className="title"
        >
          Saved Properties
        </div>
        <div
          className="block"
        />
      </div>
    </div>
  </div>
</div>
`;
